frontend/src/components/GeyserStakeView.tsx

```diff
@@ -1,42 +1,77 @@
+import { GeyserInteractionButton, Paragraph } from '../styling/styles'
+import { amountOrZero, formatAmount } from '../utils/amount'
+import { PositiveInput } from './PositiveInput'
 
-interface GeyserStakeViewProps {}
+interface Props {}
 
-export const GeyserStakeView: React.FC<GeyserStakeViewProps> = () => {
+export const GeyserStakeView: React.FC<Props> = () => {
   const [amount, setAmount] = useState<string>('')
-  const { selectedGeyser } = useContext(GeyserContext)
+  const [parsedAmount, setParsedAmount] = useState<BigNumber>(BigNumber.from('0'))
+  const { selectedGeyser, stakingTokenDecimals, stakingTokenSymbol } = useContext(GeyserContext)
   const { signer } = useContext(Web3Context)
-  const { selectedVault } = useContext(VaultContext)
+  const { selectedVault, currentLock } = useContext(VaultContext)
+  const { walletAmount, refreshWalletAmount } = useContext(WalletContext)
+
+  const userStake = BigNumber.from(amountOrZero(currentLock?.amount))
+
+  const refresh = () => {
+    setAmount('')
+    setParsedAmount(BigNumber.from('0'))
+    refreshWalletAmount()
+  }
 
   const handleStake = async () => {
-    if (selectedGeyser) {
-      const parsedAmount = await parseUnitsWithDecimals(amount, MOCK_ERC_20_ADDRESS, signer!)
+    if (selectedGeyser && signer && !parsedAmount.isZero()) {
       const geyserAddress = selectedGeyser.id
+      let tx
       if (selectedVault) {
         const vaultAddress = selectedVault.id
-        await approveDepositStake(geyserAddress, vaultAddress, parsedAmount, signer as Wallet)
+        tx = await approveDepositStake(geyserAddress, vaultAddress, parsedAmount, signer as Wallet)
       } else {
-        await approveCreateDepositStake(geyserAddress, parsedAmount, signer as Wallet)
+        tx = await approveCreateDepositStake(geyserAddress, parsedAmount, signer as Wallet)
       }
+      const receipt = await tx.wait()
+      if (receipt) refresh()
     }
   }
 
+  const formatDisplayAmount = (amount: BigNumberish) => (
+    formatAmount(amount, stakingTokenDecimals, stakingTokenSymbol)
+  )
+
   return (
-    <div>
-      <Input
-        type="number"
+    <div className="p-4 flex flex-col">
+      {parsedAmount.isZero()
+        ? <>
+            <div>Wallet balance: {formatDisplayAmount(walletAmount)}</div>
+            <div>Current stake: {formatDisplayAmount(userStake)}</div>
+          </>
+        : <>
+            <div>New wallet balance: {formatDisplayAmount(walletAmount.sub(parsedAmount))}</div>
+            <div>New stake: {formatDisplayAmount(userStake.add(parsedAmount))}</div>
+          </>}
+      <PositiveInput
         placeholder="Enter amount"
         value={amount}
-        onChange={(e) => setAmount(e.currentTarget.value)}
+        onChange={(e) => {
+          setAmount(e.currentTarget.value)
+          if (selectedGeyser && signer) {
+            setParsedAmount(parseUnits(amountOrZero(e.currentTarget.value).toString(), stakingTokenDecimals))
+          }
+        }}
```

Comment:
nit: can prolly move this move into a function

Reply:
fixed